azblob/parsing_urls.go

```diff
@@ -124,6 +134,11 @@ func (up BlobURLParts) URL() url.URL {
 
 	rawQuery := up.UnparsedParams
 
+	// Check: Both snapshot and version id cannot be present in the request URL.
+	if up.Snapshot != "" && up.VersionID != "" {
+		log.Fatal("Snapshot and versioning cannot be enabled simultaneously.")
```

Comment:
>log.Fatal("Snapshot and versioning cannot be enabled simultaneously.") [](start = 2, length = 70)

We cannot do fatal logging in any SDK: // Fatal is equivalent to Print() followed by a call to os.Exit(1).

Let's avoid this check, or return an error

Reply:
Ohh I see. Alright. I'll do `errors.New()` to throw this error.